include/caffe/datum_DB_factory.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef CAFFE_DATUMDB_FACTORY_H_
+#define CAFFE_DATUMDB_FACTORY_H_
+
+#include <map>
+#include <string>
+
+#include "caffe/common.hpp"
+#include "caffe/proto/caffe.pb.h"
+
+namespace caffe {
+
+class DatumDB;
+
+class DatumDBRegistry {
```

Comment:
I guess we could, but I made it a bit more specific to store only one `DatumDB` per source, this avoids opening twice the same DB